WooCommerce/Classes/ViewRelated/Coupons/CouponListViewModel.swift

```diff
@@ -240,7 +277,16 @@ private extension CouponListViewModel {
         if hasData {
             state = .coupons
         } else {
-            state = .empty
+            loadCouponSetting { [weak self] result in
```

Comment:
is it possible to call this only when there is an error from the call site (`handleCouponSyncResult`), if an error is thrown when coupons are disabled? since `loadCouponSetting` internally makes a network request, it might save some API requests when coupons are enabled and there are just no coupons

Reply:
This makes sense too, updated in 35116c7765602c4db7b58495b9dc22224e3ac1c8 👍